omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java

```diff
@@ -394,23 +397,28 @@ public static void setConcept(Obs obs, Object value) {
 	@PropertySetter("value")
 	public static void setValue(Obs obs, Object value) throws ParseException, ConversionException, IOException {
 		if (value != null) {
+
+			// special case for complex obs
 			if (obs.isComplex()) {
 				byte[] bytes = DatatypeConverter.parseBase64Binary(value.toString());
-				
+
 				ComplexData complexData = new ComplexData(obs.getUuid() + ".raw", new ByteArrayInputStream(bytes));
 				obs.setComplexData(complexData);
-			} else if (obs.getConcept().getDatatype().isCoded()) {
-				// setValueAsString is not implemented for coded obs (in core)
-				
-				//We want clients to be able to fetch a coded value in one rest call
-				//and set the returned payload as the obs value
+				return;
```

Comment:
I was reluctant to refactor this too much because I wasn't sure how good the test coverage was, but the nested if/else block were confusing and this change was going to make them even more confusing, so I removed the top-level if/else and had each block just return instead.